cmd/zed/lake/serve/command.go

```diff
@@ -1,93 +1,81 @@
-package listen
+package serve
 
 import (
 	"context"
 	"errors"
 	"flag"
-	"fmt"
 	"io"
 	"io/ioutil"
 	"net"
 	"os"
 	"os/signal"
 	"runtime"
 
-	"github.com/brimdata/zed/api"
 	"github.com/brimdata/zed/cli"
+	zedlake "github.com/brimdata/zed/cmd/zed/lake"
 	"github.com/brimdata/zed/pkg/charm"
 	"github.com/brimdata/zed/pkg/fs"
 	"github.com/brimdata/zed/pkg/httpd"
 	"github.com/brimdata/zed/pkg/rlimit"
-	"github.com/brimdata/zed/ppl/cmd/zqd/logger"
-	"github.com/brimdata/zed/ppl/cmd/zqd/root"
-	"github.com/brimdata/zed/ppl/zqd"
-	"github.com/brimdata/zed/proc/sort"
+	"github.com/brimdata/zed/service"
+	"github.com/brimdata/zed/service/logger"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
-	"gopkg.in/yaml.v3"
 )
 
-var Listen = &charm.Spec{
-	Name:  "listen",
-	Usage: "listen [options]",
-	Short: "listen as a daemon and repond to zqd service requests",
+var Serve = &charm.Spec{
+	Name:  "serve",
+	Usage: "serve [options]",
+	Short: "serve as a daemon and repond to zqd service requests",
 	Long: `
 The listen command launches a process to listen on the provided interface and
 `,
-	HiddenFlags: "brimfd,filestorereadonly",
+	HiddenFlags: "brimfd,filestorereadonly,nodename,podip,recruiter,workers",
 	New:         New,
 }
 
 func init() {
-	root.Zqd.Add(Listen)
+	zedlake.Cmd.Add(Serve)
 }
 
 type Command struct {
-	*root.Command
-	conf       zqd.Config
-	logger     *zap.Logger
-	loggerConf *logger.Config
+	lake    *zedlake.Command
+	conf    service.Config
+	logger  *zap.Logger
+	logconf logger.Config
 
 	// Flags
 
-	// brimfd is a file descriptor passed through by brim desktop. If set zqd
-	// will exit if the fd is closed.
+	// brimfd is a file descriptor passed through by brim desktop. If set the
+	// command will exit if the fd is closed.
 	brimfd     int
-	configfile string
 	devMode    bool
 	listenAddr string
-	logLevel   zapcore.Level
 	portFile   string
 }
 
 func New(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
-	c := &Command{Command: parent.(*root.Command)}
+	c := &Command{lake: parent.(*zedlake.Command)}
 	c.conf.Auth.SetFlags(f)
-	c.conf.DB.SetFlags(f)
-	c.conf.ImmutableCache.SetFlags(f)
```

Comment:
The cache is going to move but do we need to take it out now?  We'll only have to put it back here later.

Reply:
I can keep it in, I removed it because it needed hooks into archive that no longer exist in lake.